aws/cloudformation/standalone/gen_ai_curriculum/config.rb

```diff
@@ -0,0 +1,66 @@
+require "ostruct"
+require_relative '../../../../lib/cdo/shared_constants'
+
+module Config
+  # Configuration for each endpoint used in the Gen AI Curriculum. Required properties:
+  #   hf_model_id: HuggingFace model ID
+  #   model_id: App-wide model ID (from shared constants)
+  #   model_name: Short name for naming resources (letters only)
+  #   instance_type: EC2 Instance Type for endpoint
+  #   min_num_instances (prod only): Minimum number of deployed instances used for autoscaling on production
+  #     on test, only a single instance will be used without autoscaling to limit resource usage
+  #   max_num_instances (prod only): Maximum number of deployed instances used for autoscaling on production
+  #     not used on test
+  #   autoscaling_target_value (prod only): Number of requests / minute at which autoscaling starts to take effect
+  #     not used on test
+  ENDPOINT_CONFIGS = [
+    {
+      hf_model_id: "mistralai/Mistral-7B-Instruct-v0.1",
+      model_id: SharedConstants::AI_CHAT_MODEL_IDS[:MISTRAL],
+      model_name: "Mistral",
+      instance_type: {
+        prod: "ml.g5.4xlarge",
+        test: "ml.g5.xlarge"
+      },
+      min_num_instances: 3,
```

Comment:
@code-dot-org/infrastructure noting that we'll need to make another quota increase request before deploying to support both prod and test variants

Reply:
From the "supporting prod experimenting with this" side a quota increase would be really nice.... if we can get higher bandwidth instances stood up load-wise it would help prod run back to back e.g. thousand of message tests much quicker. On an g5.xlarge its pretty glacial, haven't tried parallelizing yet tho. Are we saturating these instances? Do you know if parallelizing gives more throughput?

Reply:
From my manual load testing, I didn't see much of a latency difference between an ml.g5.xlarge and an ml.g5.8xlarge, though I didn't do anything special to try and optimize parallelizing requests. AFAIK these instances were parallelizing as much as possible automatically (saw up to ~100 concurrent requests on a single instance). That being said though, the ml.g5.12xlarge instance which I didn't have access to at the time has 4 GPUs (as opposed to the others which have 1), so we might see some increased throughput there.

Reply:
Aha! So with ~100x (or, say, 20x) concurrent requests, the time to process each request is roughly the same as 1x?

Reply:
I just ran some tests on the ml.g5.4xlarge instances. Firstly, execution time is REALLY noisy/variable. Aren't these dedicated instances we're running? I'm really surprised by the variability.

Because of the noisiness, its hard to see the impact of parallelism, but its definitely not highly parallel beyond about 4x parallel. It helps to go higher, but the impact generally isn't huge on requests/s. It generally peaks at 15x parallel, but 15x parallel is also definitely not 3x faster than 5x parallel.

**With lots of parallelism, I'm still only seeing an ml.g5.4xlarge able to handle between 1 request/s (short responses) and 4 requests/s (very short responses).**

# Asking for a 25 word response
Ran 50 requests 5x parallel in 19.6s, averaging 19.0 words per response. 2.5 requests/s, 294.9 words/s
Ran 50 requests 10x parallel in 21.9s, averaging 23.4 words per response. 2.3 requests/s, 320.6 words/s
Ran 50 requests 15x parallel in 11.8s, averaging 18.2 words per response. 4.3 requests/s, 450.0 words/s
Ran 50 requests 20x parallel in 11.9s, averaging 24.7 words per response. 4.2 requests/s, 609.1 words/s
Ran 50 requests 25x parallel in 14.7s, averaging 31.1 words per response. 3.4 requests/s, 655.5 words/s
Ran 50 requests 30x parallel in 16.3s, averaging 24.9 words per response. 3.1 requests/s, 498.1 words/s
Ran 50 requests 50x parallel in 15.1s, averaging 21.5 words per response. 3.3 requests/s, 428.6 words/s

# Asking for a 50 word response
Ran 50 requests 5x parallel in 25.1s, averaging 38.0 words per response. 2.0 requests/s, 474.1 words/s
Ran 50 requests 10x parallel in 16.5s, averaging 37.3 words per response. 3.0 requests/s, 740.1 words/s
Ran 50 requests 15x parallel in 22.9s, averaging 57.0 words per response. 2.2 requests/s, 817.2 words/s
Ran 50 requests 20x parallel in 13.6s, averaging 41.6 words per response. 3.7 requests/s, 974.9 words/s
Ran 50 requests 25x parallel in 12.1s, averaging 49.0 words per response. 4.1 requests/s, 1307.1 words/s
Ran 50 requests 30x parallel in 20.7s, averaging 49.0 words per response. 2.4 requests/s, 740.0 words/s
Ran 50 requests 50x parallel in 18.0s, averaging 49.2 words per response. 2.8 requests/s, 916.0 words/s

# Asking for a 250 word response
Ran 50 requests 5x parallel in 116.3s, averaging 227.4 words per response. 0.4 requests/s, 615.3 words/s
Ran 50 requests 10x parallel in 74.5s, averaging 246.2 words per response. 0.7 requests/s, 1042.6 words/s
Ran 50 requests 15x parallel in 50.0s, averaging 251.5 words per response. 1.0 requests/s, 1587.4 words/s
Ran 50 requests 20x parallel in 40.5s, averaging 249.0 words per response. 1.2 requests/s, 1942.1 words/s
Ran 50 requests 25x parallel in 31.8s, averaging 247.7 words per response. 1.6 requests/s, 2500.9 words/s
Ran 50 requests 30x parallel in 29.7s, averaging 248.6 words per response. 1.7 requests/s, 2655.3 words/s
Ran 50 requests 50x parallel in 23.3s, averaging 257.6 words per response. 2.1 requests/s, 3496.2 words/s

Reply:
Yeah, my main takeaway from testing was that the kind of prompt and response was the main driver for response time. A small prompt like "hi" would usually return a response under 1 second no matter the volume of requests (until I hit rate limits) while a large input like "tell me a story about a dog" would take 15s or more. Here's a [whole bunch of data](https://docs.google.com/spreadsheets/d/11q7fP0g3AukWRrbvXprmavpaNFvWNKZ5msQDkaiVLY0/edit?gid=0#gid=0) from some load tests I did a while back if you'd like to take a look and see if it lines up with your observations